forge-game/src/main/java/forge/game/GameAction.java

```diff
@@ -264,6 +264,7 @@ private Card changeZone(final Zone zoneFrom, Zone zoneTo, final Card c, Integer
 
             if (zoneTo.is(ZoneType.Stack)) {
                 // try not to copy changed stats when moving to stack
+                copied.setChangedCardKeywords(c.getChangedCardKeywords());
```

Comment:
i still don't know if that would cause other problems, i think there was a reason why i tried to remove it

Reply:
Yea I agree but I haven't found a good alternative yet

Reply:
does it cause problems with other cards like Henzi or others what have a CastBy Keyword?

Reply:
Still unsure but it looks like your suspicion was right and #3089 did in fact introduce a regression for Henzie:
![image](https://github.com/Card-Forge/forge/assets/8506892/a9249743-25ee-4ee9-8dbf-8577f4a6a2c6)

This is on the stack, when it ETB it seems to get cleaned up again (confirmed it's not a View problem)

It doesn't always happen, but usually when you cast it directly from extrinsic Blitz

How to fix?

Reply:
it wasn't directly broken by this one i think, for the fix, i just need to store the idx inside the Keyword like i did with the static Ability id

Reply:
This part should be fixed, right?

Reply:
yea Henzie is ok

but you were right still need another solution for Niv-Mizzet
this one causes problems when combining extrinsic and intrinsic flashback

Reply:
You should try to move the other fixes into a separate MR

Reply:
Fixed it via script for now instead, there should be nothing where it matters
Another approach for an engine fix can be attempted later